### PR TITLE
feat: add new API endpoint to get and post accelerator data [DET-9748]

### PR DESCRIPTION
**Pull request by salonig23**
## Description
Added a get and post API endpoint for accelerator data for each allocation

## Test Plan
Check if endpoints show up correctly in swagger UI

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-9748

### CODE REVIEW

File: proto/src/determined/api/v1/api.proto
Comment (salonig23) [reply proto/src/determined/api/v1/api.proto]: Done!

File: proto/src/determined/api/v1/trial.proto
Comment (salonig23) [reply proto/src/determined/api/v1/trial.proto:568]: Apart from Kubernetes I think it is the same. Keeping node_name for now

File: proto/src/determined/api/v1/trial.proto
Comment (stoksc) [inline proto/src/determined/api/v1/trial.proto:534]: I would put this in task.proto instead.

**Review by stoksc — COMMENTED** [review]
lgtm modulo my comment about moving the types.

but i'll let @NicholasBlaskey have the final say.

File: proto/src/determined/api/v1/trial.proto
Comment (salonig23) [reply proto/src/determined/api/v1/trial.proto]: Done!

**Comment by salonig23** [comment]
> hey @salonig23 should I review this?
> 
> Usually I go off of if the review has been re requested on github

Please review it! I'll re-request next time!

**Review by NicholasBlaskey — APPROVED** [review]
Nice work, LGTM

**Pull request merged**
